### ISSUE TITLE
why is hist_i padding with 0?

### ISSUE THREAD

**Issue by Zhiqiang-Ye** [jinze1994/ATRank#1]
In the code /ATRank/utils/2_remap_id_raw.py, asin(item) ids are remapped to 0 ~ item_count.
In the code /ATRank/atrank/input.py, hist_i is padding with value of 0. So I guess there are two kinds of 0 in the hist_i matrix: one is real asin id 0 and the other is padding value 0.
When using tf.nn.embedding_lookup with hist_i to generate h_emb in the code /ATRank/atrank/model.py, will you fill h_emb with a lot of real item 0's vectors which do not exist in the purchase history?

Do I miss anything in your code or misunderstand any part of it?

**Comment by jinze1994** [comment]
You are right about there are two kinds of '0' in the hist_i matrix, and we mask all the padding '0' in all methods according to the sequence length `sl` of hist_i. You could find the code accordingly in model.py.
